src/Passes/Source/Rules/Factory.cpp

```diff
@@ -502,6 +507,82 @@ namespace quantum
              replace_branch_positive});
     }
 
+    void RuleFactory::optimiseSelectQuantumOne()
+    {
+        auto replace_select_positive = [](Builder& builder, Value* val, Captures& cap, Replacements& replacements) {
+            auto result = cap["result"];
+            auto cond   = llvm::dyn_cast<llvm::Instruction>(cap["cond"]);
+            // Replacing result
+            auto orig_instr = llvm::dyn_cast<llvm::Instruction>(val);
+            if (orig_instr == nullptr)
+            {
+                return false;
+            }
+
+            auto                      module   = orig_instr->getModule();
+            auto                      function = module->getFunction("__quantum__qir__read_result");
+            std::vector<llvm::Value*> arguments;
+            arguments.push_back(result);
+
+            if (!function)
+            {
+                std::vector<llvm::Type*> types;
+                types.resize(arguments.size());
+                for (uint64_t i = 0; i < types.size(); ++i)
```

Comment:
Nit: I'd suggest
```suggestion
                for (auto i = 0; i < types.size(); ++i)
```
or
```suggestion
                for (size_t i = 0; i < types.size(); ++i)
```